lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Cardano.Wallet.Byron.NetworkSpec (spec) where
+
+import Prelude
+
+import Cardano.BM.Data.Severity
+    ( Severity (..) )
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Wallet.Byron.Compatibility
+    ( NodeVersionData )
+import Cardano.Wallet.Byron.Launch
+    ( withCardanoNode )
+import Cardano.Wallet.Byron.Network
+    ( NetworkLayerLog (..), withNetworkLayer )
+import Cardano.Wallet.Network
+    ( NetworkLayer (..) )
+import Cardano.Wallet.Primitive.Types
+    ( FeePolicy (..), GenesisBlockParameters (..), TxParameters (..) )
+import Control.Retry
+    ( constantDelay, limitRetries, recoverAll )
+import Data.Maybe
+    ( mapMaybe )
+import Data.Quantity
+    ( Quantity (..) )
+import Test.Hspec
+    ( Spec, describe, it, shouldBe, shouldReturn )
+import Test.Utils.Paths
+    ( getTestData )
+import Test.Utils.Trace
+    ( withLogging )
+
+{-------------------------------------------------------------------------------
+                                      Spec
+-------------------------------------------------------------------------------}
+
+spec :: Spec
+spec = describe "getTxParameters" $ do
+    it "Correct values are queried" $ do
+        withTestNode $ \gbp sock vData -> withLogging $ \(tr, getLogs) -> do
+            let gbp' = gbp { txParameters = fakeTxParameters }
+            withNetworkLayer tr gbp' sock vData $ \nl -> do
+                let retryPolicy = constantDelay 1_000_000 <> limitRetries 10
+                recoverAll retryPolicy $ const $
+                    getTxParameters nl `shouldReturn` txParameters gbp
+            msg <- mapMaybe isMsgTxParams <$> getLogs
+            msg `shouldBe` [txParameters gbp]
```

Comment:
Feels more like an integration test than a unit test though.

Reply:
The byron network layer is not set up very well for unit testing, so using the real cardano-node was the best that I could do.